base/reflection.jl

```diff
@@ -2346,3 +2346,22 @@ function destructure_callex(topmod::Module, @nospecialize(ex))
     end
     return f, args, kwargs
 end
+
+"""
+    Base.generating_output([incremental::Bool])::Bool
+
+Return `true` if the current process is being used to pre-generate a
+code cache via any of the `--output-*` command line arguments. The
+optional argument `incremental` controls whether to only return `true`
+for that specific mode of compilation.
```

Comment:
I'm a latecomer here, but I think even some sophisticated users may not clearly understand what "incremental compilation" means. Maybe add "(i.e., `true` for package precompilation and `false` for sysimg generation)"?

Reply:
Yeah, that sounds better. I will make a PR on it.

Reply:
#51216